tests/math/sequence/test_sequential_funcs.py

```diff
@@ -46,6 +46,31 @@ def ddof_arg(*arrs):
     return st.integers(0, min_side - 1) if min_side else st.just(0)
 
 
+@st.composite
+def gen_average_args(draw, arr):
+    # Weight is either
+    # - the same shape as arr OR
+    # - 1D with compatible length
+    # See: https://github.com/numpy/numpy/blob/c31cc36a8a814ed4844a2a553454185601914a5a/numpy/lib/function_base.py#L397
+    wt_same_shape, wt_1D = range(2)
+    if arr.ndim == 0 or draw(st.one_of(st.just(wt_same_shape), st.just(wt_1D))) == wt_same_shape:
+        axis = draw(axis_arg(arr))
+        wt_shape = arr.shape
+    else: # wt_1D
+        # Only integer axis is supported for 1D weights
+        axis = draw(st.integers(min_value=0, max_value=len(arr.shape) - 1))
+        wt_shape = (arr.shape[axis],)
+    # Filter any axis summing up to 0 to avoid ZeroDivisionError
+    wt_strat = hnp.arrays(
+        dtype=np.float,
+        shape=wt_shape,
+        elements=st.floats(allow_infinity=False, allow_nan=False, min_value=-10, max_value=10),
+    ).filter(
+        lambda wt: not np.isclose( _canonicalize_weights(arr, axis, wt).sum(axis=axis), 0).any()
+    )
+    weights = draw(wt_strat | st.none())
```

Comment:
It is preferable to shrink towards simpler strategies

```suggestion
    weights = draw(st.none() | wt_strat)
```

Reply:
Interesting, where can I read more about that? I found https://hypothesis.readthedocs.io/en/latest/data.html#shrinking

Reply:
[`st.one_of`](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.one_of)

> Examples from this strategy will generally shrink to ones that come from strategies earlier in the list, then shrink according to behaviour of the strategy that produced them. In order to get good shrinking behaviour, try to put simpler strategies first. e.g. `one_of(none(), text())` is better than `one_of(text(), none())`.